utility/types/mempool_new.go

```diff
@@ -0,0 +1,107 @@
+package types
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/mempool"
+)
+
+var _ Mempool = &TxFIFOMempool{}
+
+type TxFIFOMempool struct {
+	g          *mempool.GenericFIFOSet[string, string]
+	m          sync.Mutex
```

Comment:
Mutexes are necessary but I like to avoid them whenever possible.

Provided that there is a mutex in the underlying FIFO Set, could we avoid one here? 

Could be a fun opportunity for concurrent fuzz testing

Reply:
I agree but in this case we have a state that it's split into the generic part and the custom part (where we track specific things like `txBytes` and `size`, therefore we need two mutexes.

Unless you see a cleverer way to handle this. I thought about injecting a map into the generic part but it sounded very hacky and needlessly complex